plugins/woocommerce/src/Internal/ProductAttributesLookup/Filterer.php

```diff
@@ -63,13 +63,17 @@ public function filter_by_attribute_post_clauses( array $args, \WP_Query $wp_que
 			return $args;
 		}
 
-		$clause_root = " {$wpdb->prefix}posts.ID IN (";
+		// The extra derived table ("SELECT product_or_parent_id FROM") is needed for performance
+		// (causes the filtering subquery to be executed only once).
+		$clause_root = " {$wpdb->prefix}posts.ID IN ( SELECT product_or_parent_id FROM (";
```

Comment:
For brevity, could simplify slightly to:

```suggestion
		$clause_root = " {$wpdb->posts}.ID IN ( SELECT product_or_parent_id FROM (";
```

(`$wpdb->posts` is always defined.)

Reply:
Changed, please re-review.